### PR TITLE
[astro] Added moon phase precision

### PR DESCRIPTION
**Pull request by lsiepel**
Prior to this PR the moon age was calculated in a double and rounded to a day before it is send to tthe channel.  I removed the rounding.

It has two effects:
1. The age channel (Duration) represeted in days has double value.
2. The calculation of isWaxing is more precise 

Closes https://github.com/openhab/openhab-addons/issues/6571

### CODE REVIEW

File: bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/model/MoonPhase.java [hunk 1]
```diff
@@ -105,7 +105,7 @@ public QuantityType<Time> getAge() {
     /**
      * Sets the age in days.
      */
-    public void setAge(int age) {
+    public void setAge(Double age) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/model/MoonPhase.java:108]:
```suggestion
    public void setAge(double age) {
```

**Comment by jlaur** [comment]
Just to be clear, my comment is a single comment, not a review. 😄 The problem solved here seems complex and I don't have any knowledge about the astro calculations and workings, so I hope @gerrieg will be able to review.

**Review by lolodomo — APPROVED** [review]
Code LGTM

**Comment by lolodomo** [comment]
> The problem solved here seems complex and I don't have any knowledge about the astro calculations and workings

Same for me !
But I understand that now getAge() returns a number of days which is not rounded to a day.
@lsiepel : did you check all calls of this method ?

**Comment by lsiepel** [comment]
> 
Because all the complex calculations it might look complex. But in essence it is really simple. The age was first rounded then reported and used for further calculations. More precission was added by removing the rounding.

Only side effect i can think of is that users might have rules / item labels etc. that don't expect a double for MoonAge. But that is easily fixed (if needed at all) by transforming the value to a full int. Atleast the user now has the choice to do so.

**Comment by lolodomo** [comment]
Ok, so let's merge it.

**Comment by jlaur** [comment]
> Because all the complex calculations it might look complex. But in essence it is really simple. The age was first rounded then reported and used for further calculations. More precission was added by removing the rounding.

Would you consider adding a unit test for that? :-) Only sun calculations are currently covered, but perhaps this would be an opportunity to add moon coverage.

**Comment by clinique** [comment]
Thinking of it a bit late, this can break some sitemaps using the int value to provide dynamic icon.

**Pull request merged**